python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -252,19 +247,30 @@ async def update_status(self):
                     "startAfterPowerLoss",
                     "startAfterDryerPreRun",
                 ],
-                status.registers[2],
+                status[2],
             ),
         )
 
+        self._start_by_remote = status[2] & 0x01 == 0x01
+        self._status_update = True
+
+        if status[0] & 0x02 == 0x02:
+            # None can be passed, as begin_enable and begin_disable called from
+            # _do_change_state don't care about its content
+            if self.summary_state != salobj.State.ENABLED:
+                await self.do_enable(None)
+                self.log.info("Auto switched to enabled, as compressor is running")
```

Comment:
I think this needs to be changed. You cannot automatically switch states (e.g. by calling do_enable). Normally I would expect a call to do_enable to *cause* the compressor to start running (and presumably delay the transition until it is).

Is this intended to handle the condition that the compressor was already running, or somebody switched it on manually while the CSC was in the disabled state? If so, please don't do that. Just report that it is running and leave the CSC in disabled state.

Reply:
DM-35280 solved that.